guides/source/active_support_instrumentation.md

```diff
@@ -3,623 +3,1160 @@
-  headers: #<ActionDispatch::Http::Headers:0x0055a67a519b88>,
-  format: :html,
-  method: "GET",
-  path: "/posts",
-  request: #<ActionDispatch::Request:0x00007ff1cb9bd7b8>,
-  response: #<ActionDispatch::Response:0x00007f8521841ec8>,
-  status: 200,
-  view_runtime: 46.848,
-  db_runtime: 0.157
-}
+The event is emitted when a controller action finishes processing (after filters
+and action execution, before the response is committed).
+
+For example:
+
+```ruby
+class PostsController < ApplicationController
+  def index
+    @posts = Post.limit(10)
+    render :index        # event fires as the action finishes, before commit
+  end
+end
 ```
 
+The event payload (`event.payload`) includes the following keys (with typical
+example values).
+
+| Payload Key     | Description                                               | Example                                     |
+| --------------- | --------------------------------------------------------- | ------------------------------------------- |
+| `:controller`   | The controller name                                       | `"PostsController"`                         |
+| `:action`       | The action                                                | `"index"`                                   |
+| `:params`       | Hash of request parameters without any filtered parameter | `{"controller"=>"posts","action"=>"index"}` |
+| `:headers`      | Request headers                                           | `#<ActionDispatch::Http::Headers ...>`      |
+| `:format`       | html/js/json/xml etc                                      | `:html`                                     |
+| `:method`       | HTTP request verb                                         | `"GET"`                                     |
+| `:path`         | Request path                                              | `"/posts"`                                  |
+| `:request`      | The [`ActionDispatch::Request`][] object                  | `#<ActionDispatch::Request ...>`            |
+| `:response`     | The [`ActionDispatch::Response`][] object                 | `#<ActionDispatch::Response ...>`           |
+| `:status`       | HTTP status code                                          | `200`                                       |
+| `:view_runtime` | Amount spent in view in ms                                | `46.848`                                    |
+| `:db_runtime`   | Amount spent executing database queries in ms             | `0.157`                                     |
+
 #### `send_file.action_controller`
 
-| Key     | Value                     |
-| ------- | ------------------------- |
-| `:path` | Complete path to the file |
+The event is emitted when a controller streams or sends a file with `send_file`.
 
-Additional keys may be added by the caller.
+For example:
+
+```ruby
+send_file "/var/app/exports/report.csv",
+  filename: "report.csv",
+  disposition: "attachment"
```

Comment:
I think this example would be clearer as well with a full example.
Something like:
```suggestion
class ReportsController < ApplicationController
  def download
    send_file "/var/app/exports/report.csv",
      filename: "report.csv",
      disposition: "attachment"
  end
end
```